src/test/java/com/microsoft/sqlserver/clientcertauth/ClientCertificateAuthenticationTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.clientcertauth;
+
+import static org.junit.Assert.assertTrue;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerException;
+import com.microsoft.sqlserver.jdbc.TestResource;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Constants;
+
+
+/**
+ * Tests client certificate authentication feature
+ * The feature is only supported against SQL Server Linux CU2 or higher.
+ * 
+ */
+@RunWith(JUnitPlatform.class)
+@Tag(Constants.xSQLv12)
+@Tag(Constants.xSQLv14)
+@Tag(Constants.xAzureSQLDW)
+@Tag(Constants.xAzureSQLDB)
+@Tag(Constants.clientCertAuth)
+public class ClientCertificateAuthenticationTest extends AbstractTest {
+
+    @Test
+    public void pkcs1Test() throws Exception {
+        String conStr = connectionString + ";clientCertificate=" + clientCertificate + ".pem;" + "clientKey="
+                + clientKey + "-pkcs1.key;";
+        try (Connection conn = DriverManager.getConnection(conStr); Statement stmt = conn.createStatement()) {
+            ResultSet rs = stmt.executeQuery("SELECT @@VERSION AS 'SQL Server Version'");
+            rs.next();
+            assertTrue(rs.getString(1).contains(TestResource.getResource("R_microsoft")));

```

Comment:
? why are we only checking if it contains the string Microsoft?

Reply:
Well...it can check any part of the string that @@VERSION will return. I just decided to use the keyword microsoft.

Reply:
I would suggest to use the existing `conn.isValid(timeout);` API.